### PR TITLE
fix(tiering): rename v2 + max_file_size

### PR DESCRIPTION
**Pull request by dranikpg**
* rename all V2 _v2 to regular
* add max_file_size limit

### CODE REVIEW

File: src/server/tiering/disk_storage.cc [hunk 1]
```diff
@@ -106,6 +111,10 @@ std::error_code DiskStorage::Stash(io::Bytes bytes, StashCb cb) {
   if (offset < 0) {
     size_t start = io_mgr_.Span();
     size_t grow_size = -offset;
```
Comment (kostasrim) [inline src/server/tiering/disk_storage.cc:113]: Something I noticed here, it's a `size_t` and you assigned `-offset`. Is this intented?

Comment (dranikpg) [reply src/server/tiering/disk_storage.cc:113]: yes, because the if (offset < 0) one line above 🙂

File: src/server/tiered_storage.cc [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 #include <mimalloc.h>
 
+#include <cstddef>
```
Comment (dranikpg) [inline src/server/tiered_storage.cc:9]: I really like clangd, but it's includes are so annooooying

Comment (kostasrim) [reply src/server/tiered_storage.cc:9]: `-header-insertion=never`

File: src/server/tiering/disk_storage.cc (hunk 1 above)
Comment (kostasrim) [reply src/server/tiering/disk_storage.cc:113]: I am blind :heart_eyes: :rofl:

**Review by romange — APPROVED** [review]

**Pull request merged**